### DISCUSSION TITLE
what's the difference between background job & distributed event bus

### DISCUSSION THREAD

**Discussion by ignite-404** [abpframework/abp#17910] [Q&A]
what's the difference between background job & distributed event bus
the similar api designs confuses me.
when should i use background jobs [distributed event bus]

**Comment by EngincanV** [comment]
Distributed Event Bus system allows to publish and subscribe to events that can be transferred across application/service boundaries. You can use the distributed event bus to asynchronously send and receive messages between microservices or applications.

On the other hand, Background jobs are used to queue some tasks to be executed in the background.

You can read the [Background Jobs](https://docs.abp.io/en/abp/latest/Background-Jobs) and [Distributed Event Bus](https://docs.abp.io/en/abp/latest/Distributed-Event-Bus) docs to learn more.
